tests/ui/const-generics/late-bound-vars/in_closure.rs

```diff
@@ -1,23 +1,13 @@
-// failure-status: 1
 // known-bug: unknown
-// error-pattern:internal compiler error
-// normalize-stderr-test "internal compiler error.*" -> ""
-// normalize-stderr-test "DefId\([^)]*\)" -> "..."
-// normalize-stderr-test "\nerror: internal compiler error.*\n\n" -> ""
-// normalize-stderr-test "note:.*unexpectedly panicked.*\n\n" -> ""
-// normalize-stderr-test "note: we would appreciate a bug report.*\n\n" -> ""
-// normalize-stderr-test "note: compiler flags.*\n\n" -> ""
-// normalize-stderr-test "note: rustc.*running on.*\n\n" -> ""
-// normalize-stderr-test "thread.*panicked.*:\n.*\n" -> ""
-// normalize-stderr-test "stack backtrace:\n" -> ""
-// normalize-stderr-test "\s\d{1,}: .*\n" -> ""
-// normalize-stderr-test "\s at .*\n" -> ""
-// normalize-stderr-test ".*note: Some details.*\n" -> ""
-// normalize-stderr-test "\n[ ]*\n" -> ""
-// normalize-stderr-test "compiler/.*: projection" -> "projection"
-// normalize-stderr-test ".*omitted \d{1,} frame.*\n" -> ""
-// normalize-stderr-test "error: [\s\n]*query stack during panic:\n" -> ""
-// this should run-pass
+
+// If we want this to compile, then we'd need to do something like RPITs do,
+// where nested associated constants have early-bound versions of their captured
+// late-bound vars inserted into their generics. This gives us substitutable
+// lifetimes to actually use when borrow-checking the associated const, which is
+// lowered as a totally separate body from its parent. Since this doesn't exist,
+// so we should just error rather than resolving this late-bound var with no
+// binder to actually attach it to, or worse, as a free region that can't even be
+// substituted correctly, and ICEing. - @compiler-errors
```

Comment:
The purpose of this comment is to record a more general solution than what would be needed to fix just this test and its related test (`in_closure.rs` and `simple.rs`) -- the simplest fix for just those tests would be to just go restore behavior of resolving late-bound region captured by anon consts as free regions in the parent function's body, and fix NLL to do the free region -> vid mapping correctly again (#111215 discusses this regression in the way that NLL maps free regions, cc @BoxyUwU).

But, for the record, #79298 never really correctly supported late-bound vars in general positions (i.e. other positions where the const wasn't just referencing a free region in the scope of a body), e.g.:

```rust
// ICEs even on 1.51 after #79298

#![feature(const_generics)] // `generic_const_exprs` after it was renamed.
#![allow(incomplete_features)]

const fn inner<'a: 'a>() -> usize {
    3
}

fn test<'a>() -> [u8; { inner::<'a>() }] {
    [1, 2, 3]
}

fn main() {
    test();
}
```

or in other binders, like:

```
fn test() {
    let x: dyn for<'a> Assoc<Output = [u8; { inner::<'a>() }]>;
}
```